apps/sms/services/activity/client.js

```diff
@@ -0,0 +1,62 @@
+/* global EventDispatcher,
+          threads
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Definition of all available private members.
+   * @type {Object}
+   */
+  const priv = Object.freeze({
+    client: Symbol('ActivityClient.client'),
+
+    onActivityRequest: Symbol('ActivityClient.onActivityRequest')
+  });
+
+  var ActivityClient = function() {
+    var client = this[priv.client] = threads.client(SERVICE_NAME);
```

Comment:
I divided service into two parts - client and service. It's clear why we need dedicated file for service part - because it can be used in worker/iframe, but here I experimenting with dedicated file for client to have one more abstraction around threads.js. The main reason is that I didn't want to use text method names (```client.method("my-method")```) in views especially if we have several places where client can be used as it's a bit harder to search for usages and do renames in case we need them.

But still don't have strong opinion on that, also not sure if we want singleton, factory or class here.

What do you guys think?